v1-2/learn/how-to-use-java-interoperability.md

```diff
@@ -328,6 +328,155 @@ The next sections provide more details on various aspects related to Java intero
+   
+       ...
+   }
+};
+```
+Static methods would reside outside the Ballerina object as public functions that take the name of the Java method, with the Java class name appended at the beginning as a prefix.
+ 
+Ex: A generated static method of `java.utils.UUID` class will be as follows.
+```ballerina
+public function UUID_randomUUID() returns UUID {
+ 
+   ...
+}
+```
+
+#### Fields
+All public fields of a Java class will be exposed through Ballerina bindings in the form of getters and setters. Instance fields will have the respective getters and setters inside the Ballerina object, whereas the static fields will have getters and setters outside the Ballerina object as public functions. 
+
+The getters and setters of an instance field will take the name of the field prefixed with a ‘get’ or ‘set’ at the beginning. For a static field, getters and setters (if the field is not final) will take the name of the field with a ‘get’ or ‘set’ prefix along with the Java class name appended at the beginning.
+
+#### External Interop Functions
+These interop functions take the fully qualified Java method name as the function name. However, if there exists overloaded methods, a numeric suffix will be appended at the end.
+
+#### Dependency Objects
+When there are dependent Java classes present inside generated Ballerina bindings (as parameters or return types), the bindgen tool generates an empty Ballerina object to represent each one of these classes. These would represent a Java class mapping without the constructor, method or field bindings. If one of these classes are required later, the bindgen tool could be re-run to generate the Ballerina bindings. Please note that if the complete implementation of a dependency object is generated using the bindgen tool, the existing empty Ballerina object should be manually deleted as instructed in the command output.
+
+Ex: Generated dependency object representing `java.util.List` will be as follows.
+```ballerina
+public type List object {
+ 
+   *JObject;
+  
+   public function __init(handle obj) {
+   
+       self.jObj = obj;
+   }
+};
+```
+
+#### Ballerina Object
+A Ballerina object representing a Java class will always store the handle reference of the Java class in it’s `jObj` field. To explain the implementation further, a Ballerina object representing a Java class would always be implemented using the following `JObject` abstract class, hence the handle reference could be accessed when needed.
+
+```ballerina
+public type JObject abstract object {
+ 
+   public handle jObj;
+};
+```
+
+A Ballerina object could be initialized using the `__init` function or by using a constructor. If the first approach is used, you should pass on a handle reference of the Java class to the `__init` function. This could be used if you have already obtained a handle reference through some means other than using the constructor. If not you could use the second approach to create a new object using a constructor.
```

Comment:
```suggestion
A Ballerina object could be initialized using the `__init` function or a constructor. If the first approach is used, you should pass on a handle reference of the Java class to the `__init` function. This could be used if you have already obtained a handle reference through some means other than using the constructor. If not, you could use the second approach to create a new object using a constructor.
```